### ISSUE TITLE
TypeError: Cannot read property 'category' of undefined

### ISSUE THREAD

**Issue by kikyous** [microsoft/pyright#872]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
[Error - 10:29:48 AM] An internal error occurred while while performing type checking for t.py: TypeError: Cannot read property 'category' of undefined
    at /Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:186506
    at J (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:187242)
    at Ct (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:297028)
    at /Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:236214
    at Array.forEach (<anonymous>)
    at De (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:236203)
    at /Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:188500
    at J (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:188517)
    at dt (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:279227)
    at Object.Q [as getType] (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:185913)
[Error - 10:29:48 AM] Error performing analysis: TypeError: Cannot read property 'category' of undefined
    at /Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:186506
    at J (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:187242)
    at Ct (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:297028)
    at /Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:236214
    at Array.forEach (<anonymous>)
    at De (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:236203)
    at /Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:188500
    at J (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:188517)
    at dt (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:279227)
    at Object.Q [as getType] (/Users/kikyous/.vscode/extensions/ms-python.vscode-pylance-2020.7.2/server/pyright.bundle.js:1:185913)

**To Reproduce**
use code below

**Expected behavior**
do not report error

**Screenshots or Code**

```
from datetime import datetime
class ActivityReviewedCompletenessRecorder():
    def _get_activities_need_to_calculate(self, last_calculate_time, org_ids):
        raise NotImplementedError()

    def _get_reviewed(self, activity):
        raise NotImplementedError()

    def _get_submitter(self, score):
        raise NotImplementedError()

    def _get_statistics_on_grouped_activity(self, activity):
        scores = self._get_reviewed(activity)
        user_of_group = {}
        for group in activity.group_set.groups:
            user_of_group.update({user.user_id: user.group_id for user in group.group_users})

        unique_group_ids = set()
        for score in scores:
            submitter_id = self._get_submitter(score)
            unique_group_ids.add(user_of_group[submitter_id])

        return {"reviewed": len(unique_group_ids)}

    def _get_statistics_on_personal_activity(self, activity):
        return {"reviewed": len(self._get_reviewed(activity))}

    def _get_completeness_records(self, activities):
        completeness_records = []
        created_at = datetime.utcnow()
        for activity in activities:
            if activity.submit_by_group and activity.group_set:
                stat = self._get_statistics_on_grouped_activity(activity)
            else:
                stat = self._get_statistics_on_personal_activity(activity)

            records.update(stat)
            completeness_records.append(records)

        return completeness_records
```

**VS Code extension or command-line**
ms-python.python
ms-python.vscode-pylance

**Additional context**
no

**Comment by jakebailey** [comment]
This was already fixed in 2020.7.3, see: microsoft/pylance-release#118

**Comment by kikyous** [comment]
![image](https://user-images.githubusercontent.com/818616/88128554-d2672080-cc08-11ea-8f09-23d6bcf6dbc0.png)

issue still exist in pyright 1.1.55

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright and Pylance.

@jakebailey, this is a different issue than the previous one, even though it looks similar.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.56, which I just published. It will also be in the next version of Pylance.
